tests/test_ctdb.py

```diff
@@ -563,3 +563,40 @@ def _fake_ctdb_script(pnn, recmaster):
         assert not status.is_leader()
     assert "pnn" in caplog.records[-2].getMessage()
     assert "recmaster" in caplog.records[-1].getMessage()
+
+
+def test_check_ok(tmp_path):
+    import os
+
+    datapath = tmp_path / "_ctdb"
+    datapath.mkdir()
+
+    fake_ctdb = [
+        "#!/bin/sh",
+        'if [ "$1$TESTFAIL" == "nodestatus" ]',
+        "then exit 0;",
+        "else exit 1;",
+        "fi",
+    ]
+    fake_ctdb_script = datapath / "ctdb.sh"
+    with open(fake_ctdb_script, "w") as fh:
+        fh.write("\n".join(fake_ctdb))
+        fh.write("\n")
+    os.chmod(fake_ctdb_script, 0x0755)
```

Comment:
Generally, chmod is done using octal notation.
```suggestion    
    os.chmod(fake_ctdb_script, 0o0755)
```
If this is working it might be that were just lucky as 0x0755 != 0o755

Reply:
baah brain fade!

Reply:
Found the cause for my bug. Cut paste from testjoiner() in test_join.py. Added a new commit to the PR to fix that as well.